src/Framework/MockObject/Generator.php

```diff
@@ -819,6 +819,17 @@ private function generateMock($type, $methods, $mockClassName, $callOriginalClon
                 try {
                     $method = $class->getMethod($methodName);
 
+                    // PHP 8 reports constructors in interfaces 'accurately', producing broken mocks
```

Comment:
Why not add this logic to conditional on line 822? Less copy/pasta that way.

Reply:
Alternatively modify canMockMethod to receive isInterface as an argument and put the logic there... (it's only Called in 2 places)

Reply:
Fair call, have gone with the latter.